test/test_format_spec.lua

```diff
@@ -47,4 +47,14 @@ test:test("custom pattern format with en_US locale", date_test,
               test:ok("12 October 2017" == date:format(format))
 end)
 
+test:test("segfault with format with ru_RU locale", date_test,
+          function(test, date, fields, format)
+              test:plan(1)
+              local format = icu_date.formats.pattern("d MMMM y", 'ru_RU')
```

Comment:
Could you add several tests for different locales and patterns. I guess such more issues could be catched. E.g. for chinese  locale (it's only hypothesis).

Reply:
Patch doesn't fix anything. I improved your test and after your patch it fails:

```lua
test:test("segfault with format with ru_RU locale", date_test,
          function(test, date)
              test:plan(1)

              local patterns = {
                  "d MMMM y",
                  "yyyy-MM-dd",
                  "YYYY",
                  "yyyy-MM-dd'T'HH",
                  "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ",
              }

              local locales = {
                  "af_ZA",
                  "ar_MA",
                  "br_FR",
                  "de_LI",
                  "en_CA",
                  "fr_CA",
                  "ru_RU",
                  "ta_IN",
                  "ug_CN",
                  "zh_CN",
              }

              local attempts = 50
              for _, pattern in ipairs(patterns) do
                  for _, locale in ipairs(locales) do
                      local format = icu_date.formats.pattern(pattern, locale)
                      for _ = 1, attempts do
                          date:format(format)
                      end
                      collectgarbage()
                      collectgarbage()
                  end
              end
              test:ok(true)
end)
```

Reply:
I wrote a bit different version of this patch. https://github.com/tarantool/icu-date/pull/25

Reply:
looks better, maybe close my PR and merge yours?

Reply:
Yes, seems it should be better